src/System.IO.FileSystem/src/System/IO/DirectoryInfo.cs

```diff
@@ -4,41 +4,43 @@
 
 using System.Collections.Generic;
 using System.Diagnostics;
-using System.Diagnostics.Contracts;
-using System.Security;
 
 namespace System.IO
 {
     public sealed partial class DirectoryInfo : FileSystemInfo
     {
+        private string _name;
+
         public DirectoryInfo(string path)
         {
-            if (path == null)
-                throw new ArgumentNullException(nameof(path));
-            Contract.EndContractBlock();
+            Init(originalPath: PathHelpers.ShouldReviseDirectoryPathToCurrent(path) ? "." : path,
```

Comment:
What happened to the null check?  Is that now done inside of one of these called functions instead?

Reply:
Yes, it is in Init()